### PR TITLE
Fixing EPS application when applying sqrt 

### PR DESCRIPTION
**Pull request by FedericoDiMarzo**
## Problem

- When using the pytorch_stoi loss in some training sessions some NaN values are were appearing
- One of the possible cause would be related to the sqrt operation since the gradient will lead to a singularity
- An EPS is applied in the current implementation but the order of the operations can allow for exploding gradients

# Solution

- Sum EPS just before applying sqrt

### CODE REVIEW

**Comment by mpariente** [comment]
Thanks, that's perfect !
